File: docs-examples/example-java/src/test/java/io/micronaut/nats/docs/parameters/ProductClient.java
package io.micronaut.nats.docs.parameters;

import io.micronaut.context.annotation.Requires;
// tag::imports[]
import io.micronaut.nats.annotation.NatsClient;
import io.micronaut.nats.annotation.Subject;
// end::imports[]

@Requires(property = "spec.name", value = "BindingSpec")
// tag::clazz[]
@NatsClient
public interface ProductClient {

    @Subject("product") // <1>
    void send(byte[] data);

    void send(@Subject String subject, byte[] data); // <2>
}
// end::clazz[]

File: docs-examples/example-kotlin/src/test/kotlin/io/micronaut/nats/docs/parameters/ProductClient.kt
package io.micronaut.nats.docs.parameters

import io.micronaut.context.annotation.Requires
// tag::imports[]
import io.micronaut.nats.annotation.NatsClient
import io.micronaut.nats.annotation.Subject
// end::imports[]

@Requires(property = "spec.name", value = "BindingSpec")
// tag::clazz[]
@NatsClient
interface ProductClient {

    @Subject("product") // <1>
    fun send(data: ByteArray)

    fun send(@Subject subject: String, data:ByteArray) // <2>
}
// end::clazz[]
